PersistentInbox: NPE (Cannot notify the InboxListener)

At the startup of OH (snapshot 3215):
```
2022-12-11 08:35:20.402 [INFO ] [ry.FreeboxServerDiscoveryParticipant] - Created a DiscoveryResult for Freebox Server freebox:server:xxx on IP 192.xxx.x.xxx
2022-12-11 08:35:20.933 [ERROR] [g.discovery.internal.PersistentInbox] - Cannot notify the InboxListener 'org.openhab.core.config.discovery.internal.AutomaticInboxProcessor' about a Thing ADDED event!
java.lang.NullPointerException: null
        at org.openhab.core.config.discovery.internal.PersistentInbox.setFlag(PersistentInbox.java:487) ~[?:?]
        at org.openhab.core.config.discovery.internal.AutomaticInboxProcessor.thingAdded(AutomaticInboxProcessor.java:164) ~[?:?]
        at org.openhab.core.config.discovery.internal.PersistentInbox.notifyListeners(PersistentInbox.java:512) ~[?:?]
        at org.openhab.core.config.discovery.internal.PersistentInbox.internalAdd(PersistentInbox.java:283) ~[?:?]
        at java.lang.Iterable.forEach(Iterable.java:75) ~[?:?]
        at org.openhab.core.config.discovery.internal.PersistentInbox.lambda$0(PersistentInbox.java:174) ~[?:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
        at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305) [?:?]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) [?:?]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
        at java.lang.Thread.run(Thread.java:834) [?:?]
```

--- a/bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/PersistentInbox.java
+++ b/bundles/org.openhab.core.config.discovery/src/main/java/org/openhab/core/config/discovery/internal/PersistentInbox.java
@@ -483,6 +483,8 @@ public final class PersistentInbox implements Inbox, DiscoveryListener, ThingReg
             resultImpl.setFlag((flag == null) ? DiscoveryResultFlag.NEW : flag);
             discoveryResultStorage.put(resultImpl.getThingUID().toString(), resultImpl);
             notifyListeners(resultImpl, EventType.UPDATED);
+        } else if (result == null) {
+            logger.warn("Cannot set flag for result '{}' because it can't be found in storage", thingUID);
         } else {
             logger.warn("Cannot set flag for result of instance type '{}'", result.getClass().getName());
         }

